### PR TITLE
Project: i18n preparations

### PR DESCRIPTION
**Pull request by daividssilverio**
## Why do we want this?
Code changes are there to make it easier to have a streamlined translation process.
We'll have a single source of truth for the translations, the `Resources.resx` file.

The new documentations, issue and pull request templates, together with the new [GitHub project](https://github.com/toggl/mobileapp/projects/74) are there to make it possible for anyone to contribute with translations for the Toggl apps.

## How is it done?
The iOS and Droid projects had all of their static text entries moved from views and other native solutions (xibs, storyboards, xmls) to the `Resources.resx` file.

The language is loaded during app startup in the `AppStart`.

### Side effects
It's gonna be more complicated to make releases and make text changes now.
Everybody is gonna have to be more careful when making changes that affect texts.

## Review hints
Review by commit.
If you want to test if the translations work, please add a translated file, `Resources.ja.resx` for example, prefixing every text entry or something like that, change the device preferred language to the language you've translated and check the results. Yes, the app must be restarted for the translation to be kick in.

## :squid: Permissions
Merge, don't squash. Only me.

### CODE REVIEW

File: Toggl.Droid/Activities/AboutActivity.cs [hunk 1]
```diff
@@ -40,5 +40,11 @@ protected override void InitializeBindings()
                 .BindAction(ViewModel.OpenTermsOfServiceView)
                 .DisposedBy(DisposeBag);
         }
+
+        public override void Finish()
```
Comment (heytherewill) [inline Toggl.Droid/Activities/AboutActivity.cs]: This override is not needed

File: Toggl.Droid/Activities/CalendarSettingsActivity.cs [hunk 2]
```diff
@@ -28,7 +26,7 @@ public CalendarSettingsActivity(IntPtr javaReference, JniHandleOwnership transfe
             : base(javaReference, transfer)
         {
         }
-
+
```
Comment (heytherewill) [inline Toggl.Droid/Activities/CalendarSettingsActivity.cs]: Extra white space

File: Toggl.Droid/Activities/EditDurationActivity.cs [hunk 3]
```diff
@@ -56,7 +55,7 @@ public EditDurationActivity(IntPtr javaReference, JniHandleOwnership transfer)
             : base(javaReference, transfer)
         {
         }
-
+
```
Comment (heytherewill) [inline Toggl.Droid/Activities/EditDurationActivity.cs]: Extra white space

File: Toggl.Droid/Activities/EditProjectActivity.cs [hunk 4]
```diff
@@ -35,7 +34,7 @@ public EditProjectActivity(IntPtr javaReference, JniHandleOwnership transfer)
         protected override void InitializeBindings()
         {
             errorText.Visibility = ViewStates.Gone;
-
+
```
Comment (heytherewill) [inline Toggl.Droid/Activities/EditProjectActivity.cs]: This view shouldn't have been changed

File: Toggl.Droid/Activities/EditTimeEntryActivity.cs [hunk 5]
```diff
@@ -14,10 +12,10 @@
 using Toggl.Core.UI.ViewModels;
 using Toggl.Droid.Extensions;
 using Toggl.Droid.Extensions.Reactive;
-using Toggl.Droid.Presentation;
-using Toggl.Shared;
 using Toggl.Shared.Extensions;
-using static Toggl.Droid.Resource.String;
+using TagsAdapter = Toggl.Droid.Adapters.SimpleAdapter<string>;
```
Comment (heytherewill) [inline Toggl.Droid/Activities/EditTimeEntryActivity.cs]: I don't think these usings are correct

File: Toggl.Droid/Activities/MainTabBarActivity.cs [hunk 6]
```diff
@@ -1,7 +1,6 @@
 ﻿using Android.App;
 using Android.Content;
 using Android.Content.PM;
-using Android.OS;
```
Comment (heytherewill) [inline Toggl.Droid/Activities/MainTabBarActivity.cs]: This file shouldn't have changed

File: Toggl.Droid/Activities/SelectClientActivity.cs [hunk 7]
```diff
@@ -25,8 +23,7 @@ public partial class SelectClientActivity : ReactiveActivity<SelectClientViewMod
 
         public SelectClientActivity(IntPtr javaReference, JniHandleOwnership transfer)
             : base(javaReference, transfer)
-        {
-        }
+        { }
```
Comment (heytherewill) [inline Toggl.Droid/Activities/SelectClientActivity.cs]: No need to change this file

File: Toggl.Droid/Activities/SelectCountryActivity.cs [hunk 8]
```diff
@@ -2,8 +2,6 @@
 using Android.Content.PM;
 using Android.Runtime;
 using System;
-using System.Linq;
-using System.Reactive.Linq;
```
Comment (heytherewill) [inline Toggl.Droid/Activities/SelectCountryActivity.cs]: No need to change this file

File: Toggl.Droid/Activities/SelectTagsActivity.cs [hunk 9]
```diff
@@ -28,8 +24,7 @@ public partial class SelectTagsActivity : ReactiveActivity<SelectTagsViewModel>
 
         public SelectTagsActivity(IntPtr javaReference, JniHandleOwnership transfer)
             : base(javaReference, transfer)
-        {
-        }
+        { }
```
Comment (heytherewill) [inline Toggl.Droid/Activities/SelectTagsActivity.cs]: No changes needed here

File: Toggl.Droid/Activities/TokenResetActivity.cs [hunk 10]
```diff
@@ -2,7 +2,6 @@
 using Android.Content.PM;
 using Android.Runtime;
 using System;
-using System.Reactive.Linq;
```
Comment (heytherewill) [inline Toggl.Droid/Activities/TokenResetActivity.cs]: No need to touch this file

File: Toggl.Droid/Fragments/CalendarPermissionDeniedFragment.cs [hunk 11]
```diff
@@ -19,6 +19,7 @@ public override View OnCreateView(LayoutInflater inflater, ViewGroup container,
         public override void OnViewCreated(View view, Bundle savedInstanceState)
         {
             base.OnViewCreated(view, savedInstanceState);
+
```
Comment (heytherewill) [inline Toggl.Droid/Fragments/CalendarPermissionDeniedFragment.cs]: No need to touch this file

File: Toggl.Droid/Fragments/NoWorkspaceFragment.cs [hunk 12]
```diff
@@ -33,6 +33,7 @@ public override View OnCreateView(LayoutInflater inflater, ViewGroup container,
         public override void OnViewCreated(View view, Bundle savedInstanceState)
         {
             base.OnViewCreated(view, savedInstanceState);
+
```
Comment (heytherewill) [inline Toggl.Droid/Fragments/NoWorkspaceFragment.cs]: No need to touch this file

File: Toggl.Droid/Resources/values/Strings.xml [hunk 13]
```diff
@@ -1,160 +1,6 @@
...
-    <string name="NoEmailHandlerErrorTitle">Oops! Seems like your email account is not set up</string>
-    <string name="NoEmailHandlerErrorMessage">Set up an email account or send us your feedback to support@toggl.com</string>
-    <string name="OnboardingGetStarted">Let\'s get started!</string>
-    <string name="OnboardingKickstartProductivity">Kickstart your productivity\nwith Toggl</string>
-    <string name="OutdatedAppMessage">Seems you haven’t updated the app
-for quite some time… Please update
-your app to continue.</string>
-    <string name="TryTogglDotCom"> Try Toggl.com </string>
-    <string name="UpdateTheApp">Update the app</string>
     <string name="Reports">Reports</string>
```
Comment (heytherewill) [inline Toggl.Droid/Resources/values/Strings.xml]: Why do we still have strings here?

File: Toggl.Droid/Resources/values/Strings.xml [hunk 14]
```diff
@@ -1,160 +1,6 @@
...
     <string name="NewTimeEntry">New time entry</string>
```
Comment (heytherewill) [inline Toggl.Droid/Resources/values/Strings.xml]: I get app name, but what about these?

File: Toggl.Droid/Toggl.Droid.csproj [hunk 15]
```diff
@@ -46,7 +46,6 @@
     <AndroidSigningStorePass>{KEYSTORE_PASSWORD}</AndroidSigningStorePass>
     <AndroidSigningKeyAlias>{KEYSTORE_ALIAS}</AndroidSigningKeyAlias>
     <AndroidSigningKeyPass>{KEYSTORE_ALIAS_PASSWORD}</AndroidSigningKeyPass>
-    <AndroidLinkMode>SdkOnly</AndroidLinkMode>
```
Comment (heytherewill) [inline Toggl.Droid/Toggl.Droid.csproj]: Why?

File: docs/localization.md [hunk 16]
```diff
@@ -21,12 +21,14 @@ _Note: The string in the IB will be replaced, so it's better to make it somethin
...
-    android:id="@+id/OopsTextView"
-    android:textSize="13sp"
-    android:textAllCaps="true"
-/>
-```
+### 💻 For programmatically built views
+
+- Setup the UI's text using the `System.Resources` namespace, much like iOS.
+
+### 📐 or Xml-based views
```
Comment (heytherewill) [inline docs/localization.md]: For*

File: Toggl.Droid/Activities/EditTimeEntryActivity.cs (hunk 5 above)
Comment (daividssilverio) [reply Toggl.Droid/Activities/EditTimeEntryActivity.cs]: You have introduced this on #5767.

File: Toggl.Droid/Resources/values/Strings.xml (hunk 13 above)
Comment (daividssilverio) [reply Toggl.Droid/Resources/values/Strings.xml]: Yeah, `TileServices` need them.

File: Toggl.Droid/Resources/values/Strings.xml (hunk 14 above)
Comment (daividssilverio) [reply Toggl.Droid/Resources/values/Strings.xml]: Tile Services.

File: Toggl.Droid/Toggl.Droid.csproj (hunk 15 above)
Comment (daividssilverio) [reply Toggl.Droid/Toggl.Droid.csproj]: It looks like a bad merge. I'll re add it.

File: docs/localization.md (hunk 16 above)
Comment (daividssilverio) [reply docs/localization.md]: No, it's `or`, it's following the iOS pattern.
## ☀️ iOS
### 💻 For programatically built views
### 📐 or Interface Builder (xib, storyboard) views

## 🤖 Droid
### 💻 For programmatically built views
### 📐 or Xml-based views

File: Toggl.Droid/Resources/values/Strings.xml (hunk 13 above)
Comment (heytherewill) [reply Toggl.Droid/Resources/values/Strings.xml]: And I assume TileServices can't use .net stuff, so this file will still require translation, correct?

Comment (daividssilverio) [reply Toggl.Droid/Resources/values/Strings.xml]: Yes. You are correct.

**Pull request merged**